app/ports/management/commands/update-rsync.py

```diff
@@ -0,0 +1,46 @@
+import subprocess
+import json
+
+from django.core.management.base import BaseCommand, CommandError
+
+from parsing_scripts import git_update
+from ports.models import Port, Commit
+
+RSYNC = "rsync://rsync.macports.org/macports//trunk/dports/PortIndex_darwin_16_i386/PortIndex.json"
+JSON_FILE = "portindex.json"
+
+
+class Command(BaseCommand):
+    help = "Populates the database with Initial data from portindex.json file"
+
+    def add_arguments(self, parser):
+        parser.add_argument('new',
+                            nargs='?',
+                            default=False,
+                            help="Define a commit till which the update should be processed")
+        parser.add_argument('old',
+                            nargs='?',
+                            default=False,
+                            help="Not recommended. Helps you provide a commit from which update should start")
+
+    def handle(self, *args, **options):
+        # fetch from rsync
+        subprocess.call(['/usr/bin/rsync', RSYNC, JSON_FILE])
+
+        # Open the file
+        with open(JSON_FILE, "r", encoding='utf-8') as file:
+            data = json.load(file)
+
+        if options['new'] is False:
+            ports_to_be_updated = git_update.get_list_of_changed_ports(data['info']['commit'], False)
+        else:
+            ports_to_be_updated = git_update.get_list_of_changed_ports(options['new'], options['old'])
```

Comment:
I do have some hard time following the logic. It kind of looks as if `options['new']` was commit it, but it seems to be a boolean value. It's possible to figure it out eventually, but it could be somewhat clearer.

Reply:
If a commit hash is not supplied to the command then `options['new']` is `False` by default.